crates/amaru-consensus/src/consensus/header_validation.rs

```diff
@@ -32,12 +32,14 @@ pub fn assert_header<'a>(
 
     if let Some(epoch_nonce) = epoch_to_nonce.get(&epoch) {
         // TODO: Take this parameter from an input context, rather than hard-coding it.
-        let active_slots_coeff: FixedDecimal =
-            FixedDecimal::from(5u64) / FixedDecimal::from(100u64);
+        let active_slot_coeff: FixedDecimal = FixedDecimal::from(5u64) / FixedDecimal::from(100u64);
 
-        let block_validator = BlockValidator::new(header, ledger, epoch_nonce, &active_slots_coeff);
-
-        block_validator.validate().or_panic()
+        praos::header::assert_all(header, ledger, *epoch_nonce, &active_slot_coeff)
+            .and_then(|assertions| {
+                use rayon::prelude::*;
+                assertions.into_par_iter().try_for_each(|assert| assert())
+            })
+            .or_panic()
```

Comment:
Note: parallel running of the assertions has been moved here from downstream dependency. Ideally, as discussed, this should end up in the amaru crate. But, baby steps.

Reply:
I am not sure I understand why you want to expose the details of header validation at the level of the `amaru` exe. Or do you mean the decision to evaluate sequentially or concurrently?

Reply:
I mean the latter indeed. And I am open to suggestions as for how you would imagine that without leaking internals or exposing "Amaru" to the header validation directly.